setup.py

```diff
@@ -39,30 +40,32 @@
 ] + tornado_requires
 
 
-cmdclass = {}
-ext_modules = []
-
 # pypy detection
 PYPY = "__pypy__" in sys.modules
 UNIX = platform.system() in ("Linux", "Darwin")
 
-# only build ext in CPython with UNIX platform
+ext_modules = []
 if UNIX and not PYPY:
-    from Cython.Build import cythonize
-    cythonize("thriftpy2/transport/cybase.pyx")
-    cythonize("thriftpy2/transport/**/*.pyx")
-    cythonize("thriftpy2/protocol/cybin/cybin.pyx")
-
-    ext_modules.append(Extension("thriftpy2.transport.cybase",
-                                 ["thriftpy2/transport/cybase.c"]))
-    ext_modules.append(Extension("thriftpy2.transport.buffered.cybuffered",
-                                 ["thriftpy2/transport/buffered/cybuffered.c"]))
-    ext_modules.append(Extension("thriftpy2.transport.memory.cymemory",
-                                 ["thriftpy2/transport/memory/cymemory.c"]))
-    ext_modules.append(Extension("thriftpy2.transport.framed.cyframed",
-                                 ["thriftpy2/transport/framed/cyframed.c"]))
-    ext_modules.append(Extension("thriftpy2.protocol.cybin",
-                                 ["thriftpy2/protocol/cybin/cybin.c"]))
+    ext_modules = [
+        Extension("thriftpy2.transport.cybase", sources=["thriftpy2/transport/cybase.pyx"]),
+        Extension("thriftpy2.transport.buffered.cybuffered", sources=["thriftpy2/transport/buffered/cybuffered.pyx"]),
+        Extension("thriftpy2.transport.framed.cyframed", sources=["thriftpy2/transport/framed/cyframed.pyx"]),
+        Extension("thriftpy2.transport.memory.cymemory", sources=["thriftpy2/transport/memory/cymemory.pyx"]),
+        Extension("thriftpy2.transport.sasl.cysasl", sources=["thriftpy2/transport/sasl/cysasl.pyx"]),
+        Extension("thriftpy2.protocol.cybin", sources=["thriftpy2/protocol/cybin/cybin.pyx"]),
+    ]
+
+
+class CustomBuildExtCommand(build_ext):
```

Comment:
Hi, I want to know that's the proposal for change the setup process?

The pip / setuptools have changed so much now days, and we specified the cython dependency in `pyproject.toml`. So in the `pip install` (or other modern python package mangagement tools), the cython will be installed automatically. So I think the `build_ext` command is not needed.

Reply:
My issue was that I was running `setup.py bdist_wheel` (I believe, will have to double-check) in an empty venv on a Unix system. That broke because setup.py expects Cython to be pre-installed. By adding the `CustomBuildExtCommand`, setuptools has time to install Cython as a build requirement before building the extension.

Reply:
For local development, use `pip install -e .` in a venv (or modern tool like rye / pipenv). The `python setup.py ...` way is to be eliminated.

Reply:
I just tested with `python3 -m build --wheel` on the main branch and that works without my changes.

I still think that adding a `CustomBuildExtCommand` is the "correct" way to create extension modules, but I can revert the patch to keep your approach and only add the new sasl transport source file.

Reply:
I think this is not related to the sasl transport implementation, so I suggest just revert it, and maybe we can discuss about it in a separate thread.